### PR TITLE
 Create a Service for providing video/category data #10

### PR DESCRIPTION
**Pull request by OGTor**
# New Feature Submissions:

This will add Services for providing video/category data

Data can be is reusable for different components. 

### To Test
Open project and use 
npm run web
open the console in your desired web-browser
you will se different console logs with Category Labels, Videos, etc

### CODE REVIEW

**Comment by OGTor** [comment]
## Updated upon reviews
#### Link to commited changes
https://ogtor.github.io/web-client/
## 
image

**Review by tryklick — APPROVED** [review]

**Comment by ar9708** [comment]
FYI, checkboxes to tick @OGTor & @tryklick :

image

**Review by ar9708 — CHANGES_REQUESTED** [review]
FYI, checkboxes to tick @OGTor & @tryklick :

image

TODOs:
- [x] Linear history
- [x] Signed commints

**Review by OGTor — COMMENTED** [review]
Final fix for testData.json. 
This update should fix the problem with Actions not loading testData.json

**Comment by OGTor** [comment]
> Final fix for testData.json. This update should fix the problem with Actions not loading testData.json

https://ogtor.github.io/web-client/

image

**Review by OGTor — COMMENTED** [review]
in videoServices.test.tsx it seems as this code part (line 47 - 50)
```
    type Category = {
      id: number;
      label: string;
    };
    
```
is not used in the testing

**Review by tryklick — APPROVED** [review]

File: OwnTube.tv/components/videosOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          <View>
+            {videos
+              .filter((video) => video.category.label === category)
+              .map((video) => (
+                <View key={video.id}>
+                  <Text>ID: {video.id}</Text>
+                  <Text>Name: {video.name}</Text>
+                  <Text>Category ID: {video.category.id}</Text>
+                  <Text>Category Label: {video.category.label}</Text>
+                  <Image source={{ uri: video.thumbnailUrl }} />
```
Comment (ar9708) [inline OwnTube.tv/components/videosOverview.tsx:63]: It looks right'ish, but when looking at the page here then there are no thumbnails rendered: https://ogtor.github.io/web-client/

Do you know why?

File: OwnTube.tv/types.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  readonly category: { id: number | null; label: string };
+  readonly thumbnailPath: string;
+  readonly thumbnailUrl?: string;
+  readonly description?: string | null;
+}
+
+export interface Category {
+  readonly label: string;
+  readonly id: number;
+}
```
Comment (ar9708) [inline OwnTube.tv/types.ts:13]: I'm a bit unsure about specifying things as `readonly` here.. Let's talk about it during stand-up today.

**Review by ar9708 — COMMENTED** [review]
Good job! Starting to look neat and crispy. 🌟 Please see my comments below.

File: OwnTube.tv/types.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+export interface VideoServiceState {
+  videos: Video[];
+  categories: string[];
+  error: string | null;
+}
```
Comment (ar9708) [inline OwnTube.tv/types.ts:19]: Not a shared type, move it to `components/videosOverview` please!

**Review by ar9708 — CHANGES_REQUESTED** [review]
@OGTor Your GitHub Actions pipeline is failing, please address!

**Review by ar9708 — APPROVED** [review]
Great, looking good! 🙃

**Pull request merged**